### PR TITLE
Update native code creating a flat map proof [ECR-2313]

### PR DESCRIPTION
**Pull request by dmitry-timofeev**
## Overview

Update the native code converting Rust map proof into Java proof.

**Assumes that it's integrated after #469 . Use ca0ec3b31375af5c0467774bee725202753818d9 if need to use _before_**

---
See: https://jira.bf.local/browse/ECR-2313

### Definition of Done

- [x] There are no TODOs left in the code
- [ ] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [ ] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has Javadoc
- [x] Method preconditions are checked and documented in the Javadoc of the method
- [ ] The [continuous integration build](https://www.travis-ci.org/exonum/exonum-java-binding) passes

### CODE REVIEW

**Comment by dmitry-timofeev** [comment]
> create_java_map_proof_entries vs create_java_map_entries naming seems confusing to me
Yes :upside_down_face: 

create_java_proof_nodes (as tree nodes in a contour)? Would you suggest anything?

**Comment by dmitry-timofeev** [comment]
@vitvakatu I have some troubles with updating to the rustfmt-preview — cargo won't pick it up — would you reformat the thing, please?

**Comment by vitvakatu** [comment]
@dmitry-timofeev done

File: exonum-java-binding-core/rust/src/storage/proof_map_index.rs [hunk 1]
```diff
@@ -179,86 +178,111 @@ pub extern "system" fn Java_com_exonum_binding_storage_indices_ProofMapIndexProx
...
-    let value: JObject = env.byte_array_from_slice(value)?.into();
-    env.new_object(
-        MAP_PROOF_ENTRY_LEAF,
-        "([B[B)V",
-        &[proof_path.into(), value.into()],
-    )
+    let entries: Vec<(&Key, &Value)> = checked_proof.entries();
+    let java_entries = env.new_object_array(entries.len() as jsize, MAP_ENTRY, JObject::null())?;
+    for (i, (key, value)) in entries.iter().enumerate() {
+        // todo: Estimate precisely the upper bound on the number of references ^ and
```
[the remark this answers is not in the document]
Comment (dmitry-timofeev) [reply exonum-java-binding-core/rust/src/storage/proof_map_index.rs]: I confused the forks, sorry

File: exonum-java-binding-core/rust/src/storage/proof_map_index.rs [hunk 2]
```diff
@@ -32,14 +33,12 @@ use JniResult;
 type Key = [u8; PROOF_MAP_KEY_SIZE];
 type Index<T> = ProofMapIndex<T, Key, Value>;
 
-const MAP_PROOF_ENTRY_BRANCH: &str =
-    "com/exonum/binding/storage/proofs/map/flat/MapProofEntryBranch";
-const MAP_PROOF_ENTRY_LEAF: &str = "com/exonum/binding/storage/proofs/map/flat/MapProofEntryLeaf";
 const MAP_PROOF_ENTRY: &str = "com/exonum/binding/storage/proofs/map/flat/MapProofEntry";
+const MAP_ENTRY: &str = "com/exonum/binding/storage/proofs/map/flat/MapEntry";
```
[the remark this answers is not in the document]
Comment (vitvakatu) [reply exonum-java-binding-core/rust/src/storage/proof_map_index.rs]: Ah, sorry

**Comment by dmitry-timofeev** [comment]
Thanks everyone, merged.

**Pull request merged**